execution-engine/src/fs.rs

```diff
@@ -1049,18 +1478,58 @@ impl FileSystem {
         Ok(rst)
     }
 
+    /// Read all files recursively on path `path`.
+    /// The `principal` must have the right on `path_open`,
+    /// `fd_read` and `fd_seek`.
+    pub fn read_all_files_by_absolute_path<T: AsRef<Path>>(
+        &mut self,
+        principal: &Principal,
+        path: T,
+    ) -> Result<Vec<(PathBuf, Vec<u8>)>, ErrNo> {
+        let path = path.as_ref();
+        info!("read_all_files_by_filename: {:?}", path);
+        // Convert the absolute path to relative path and then find the inode
+        let (_, inode_impl) = self.get_inode_by_inode_path(
+            &Self::ROOT_DIRECTORY_INODE,
+            path.strip_prefix("/").unwrap_or(path).clone(),
+        )?;
+        let mut rst = Vec::new();
+        if inode_impl.is_dir() {
+            for (_, sub_relative_path) in inode_impl.read_dir(&self)?.iter() {
+                let sub_relative_path =
+                    PathBuf::from(OsString::from_vec(sub_relative_path.to_vec()));
+                // Ignore the path for current and parent directories.
+                if sub_relative_path != PathBuf::from(".")
```

Comment:
These fixed paths for the parent and current directories can probably be made constants.